addons/product/models/product_pricelist_item.py

```diff
@@ -177,22 +177,46 @@ def _compute_name_and_price(self):
                     item.env, item.fixed_price, monetary=True, dp="Product Price", currency_obj=item.currency_id)
             elif item.compute_price == 'percentage':
                 percentage = self._get_integer(item.percent_price)
-                item.price = _("%s %% discount", percentage)
-            else:
-                discount_type, percentage = self._get_displayed_discount(item)
-                if not (surcharge := item.price_surcharge):
+                if item.base_pricelist_id:
                     item.price = _(
-                        "%(percentage)s %% %(discount_type)s",
+                        "%(percentage)s %% discount on %(pricelist)s",
                         percentage=percentage,
-                        discount_type=discount_type,
+                        pricelist=item.base_pricelist_id.display_name
                     )
                 else:
                     item.price = _(
-                        "%(percentage)s %% %(discount_type)s and %(price)s surcharge",
-                        percentage=percentage,
-                        price=surcharge,
-                        discount_type=discount_type,
+                        "%(percentage)s %% discount on sales price",
+                        percentage=percentage
                     )
+            else:
+                base_str = ""
+                if item.base == 'list_price':
+                    base_str = "on sales price"
+                elif item.base == 'standard_price':
+                    base_str = "on product cost"
+                elif item.base == 'pricelist' and item.base_pricelist_id:
+                    base_str = "on %(pricelist)s" % {
+                        'pricelist': item.base_pricelist_id.display_name
+                    }
+
+                extra_fee_str = ""
+                if item.price_surcharge != 0:
+                    sign = "+" if item.price_surcharge > 0 else "-"
+                    fee_str = "extra fee" if item.price_surcharge > 0 else "rebate"
+                    extra_fee_str = "%(sign)s %(amount)s %(fee)s" % {
+                        'sign': sign,
+                        'amount': formatLang(
+                            item.env, abs(item.price_surcharge), monetary=True,
+                            currency_obj=item.currency_id),
```

Comment:
```suggestion
                        'amount': formatLang(
                            env=item.env,
                            value=abs(item.price_surcharge),
                            monetary=True,
                            currency_obj=item.currency_id,
                        ),
```
One attribute per line